Scripting API documentation (#14725)
mdn/content: files/en-us/mozilla/add-ons/webextensions/api/scripting/unregistercontentscripts/index.md

---
title: scripting.unregisterContentScripts()
slug: Mozilla/Add-ons/WebExtensions/API/scripting/unregisterContentScripts
tags:
  - API
  - Add-ons
  - Extensions
  - Method
  - Reference
  - WebExtensions
  - unregisterContentScripts
  - tabs
browser-compat: webextensions.api.scripting.unregisterContentScripts
---
{{AddonSidebar()}}

Unregisters one or more content scripts.

> **Note:** This method is available in Manifest V3 or higher.

> This is an asynchronous function that returns a [`Promise`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise).

Reviewer: ```suggestion
 To use this API you must have the `"scripting"` [permission](/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/permissions) and permission for the page's URL, either explicitly as a [host permission](/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/permissions#host_permissions) or using the [activeTab permission](/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/permissions#activetab_permission).

This is an asynchronous function that returns a [`Promise`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise).
```

## Syntax

```js

> let removing = browser.scripting.unregisterContentScripts(

Reviewer: ```suggestion
await browser.scripting.unregisterContentScripts(
```

---
title: scripting.unregisterContentScripts()
slug: Mozilla/Add-ons/WebExtensions/API/scripting/unregisterContentScripts
tags:
  - API
  - Add-ons
  - Extensions
  - Method
  - Non-standard
  - Reference
  - WebExtensions
  - unregisterContentScripts
  - tabs
browser-compat: webextensions.api.scripting.unregisterContentScripts
---
{{AddonSidebar()}}

Unregisters one or more content scripts.

> **Note:** This method is available in Manifest V3 or higher.

This is an asynchronous function that returns a [`Promise`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise).

## Syntax

```js
let removing = browser.scripting.unregisterContentScripts(
  scripts         // object
)
```

### Parameters

- `scripts`{{optional_inline}} 

>   - : `array` of {{WebExtAPIRef("scripting.ContentScriptFilter")}}. A filter to identify the dynamic content scripts to unregistered. If not specified, all dynamic content scripts are unregistered.

Reviewer: `scripts` is actually an object of type `ContentScriptFilter`.

### Return value

A [`Promise`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise) that fulfills with no arguments when all the scripts are unregistered. If any error occurs, the promise fulfills with an error message.

## Examples

> This example unregisters a script the ID `a-script`:

Reviewer: ```suggestion
This example unregisters a content script that was previously registered with ID `a-script`:
```

```js

> const scriptFilter = {
>   id: "a-script"
>   };
> 
> await browser.scripting.unregisterContentScripts([scriptFilter]);

Reviewer: ```suggestion
await browser.scripting.unregisterContentScripts({
  ids: ["a-script"],
});
```

---
title: scripting.unregisterContentScripts()
slug: Mozilla/Add-ons/WebExtensions/API/scripting/unregisterContentScripts
tags:
  - API
  - Add-ons
  - Extensions
  - Method
  - Non-standard
  - Reference
  - WebExtensions
  - unregisterContentScripts
  - tabs
browser-compat: webextensions.api.scripting.unregisterContentScripts
---
{{AddonSidebar()}}

Unregisters one or more content scripts.

> **Note:** This method is available in Manifest V3 or higher.

This is an asynchronous function that returns a [`Promise`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise).

## Syntax

```js
let removing = browser.scripting.unregisterContentScripts(
  scripts,         // object
  callback         // function
)
```

### Parameters

- `scripts`{{optional_inline}} 

>   - : {{WebExtAPIRef("scripting.ContentScriptFilter")}}. A filter to identify the dynamic content scripts to unregistered. If not specified, all dynamic content scripts are unregistered.

Reviewer 2: ```suggestion
  - : `array` of {{WebExtAPIRef("scripting.ContentScriptFilter")}}. A filter to identify the dynamic content scripts to unregistered. If not specified, all dynamic content scripts are unregistered.
```
